backend/app/service/runtime.go

```diff
@@ -441,7 +441,7 @@ func (r *RuntimeService) Update(req request.RuntimeUpdate) error {
 			return err
 		}
 		go buildRuntime(runtime, imageID, req.Rebuild)
-	case constant.RuntimeNode, constant.RuntimeJava, constant.RuntimeGo, constant.RuntimePython, constant.RuntimeDoNet:
+	case constant.RuntimeNode, constant.RuntimeJava, constant.RuntimeGo, constant.RuntimePython, constant.RuntimeDotNet:
 		runtime.Version = req.Version
 		runtime.CodeDir = req.CodeDir
 		runtime.Port = req.Port
```

Comment:
There are no known irregularities, issues, or optimization suggestions present in the code snippet provided. The current implementation appears to be functioning correctly.

The code checks for an existence of certain types of image files specified by the case statement (constant) and handles situations where they do and don't exist using switch cases respectively. Also, it validates various conditions related to paths/files such as checking if directories exists. Additionally, some functions like handlePHP() and checkPortExist() also perform their specific tasks efficiently while maintaining readability. Overall, there doesn't seem to be anything requiring major changes here.

For efficiency improvements, you might consider implementing caching mechanisms for frequently accessed files/directories or performing validations that require less overhead without impacting performance too much. However, such refinements would usually occur only when necessary after comprehensive testing has established the current function’s reliability and utility under different scenarios.